hips/hip-00XX.md

```diff
@@ -0,0 +1,213 @@
+However, given this different semantics of `--force`, a new flag is required)
+
+An additional special case is rollbacks on failed upgrade (`--atomic` flag). It is possible for Helm to attempt to rollback to a version which now has a conflict.
+For which the rollback will fail.
+Since rollbacks are "best effort" and can fail for other reasons as well, it seems best to treat this new rollback failure-case similarly (report to the user via logging and error).
+If the user specified `--force-conflicts` to the upgrade, the corresponding rollback should similarly force conflicts (this is the behavior of `--force`).
+If the rollback is to a release version which previously didn't have SSA enabled, the rollback install should also disable SSA.
+
+#### Special
+
+- `--dry-run=client` won't work with SSA enabled
+- If the user attempts to use SSA with a cluster which does not support it (unlikely: SSA [went GA](https://kubernetes.io/blog/2021/08/06/server-side-apply-ga/) in Kubernetes v1.22), Helm should error
+- Client-side API object validation isn't applicable when SSA enabled (`helm install|upgrade --disable-openapi-validation`)
+- Hooks could be deployed with SSA. But as hook objects are not updated (create/delete only), they are not really affected
+- Stored release manifests should not change (they store the object as Helm intends)
+
+### Implementation
+
+Helm will send object's manifests to the Kubernetes API server via SSA patches. Similar to `kubectl apply`.
+Utilizing the field manager name `"helm"`:
+
+https://github.com/kubernetes/kubectl/blob/197123726db24c61aa0f78d1f0ba6e91a2ec2f35/pkg/cmd/apply/apply.go#L248
+https://github.com/kubernetes/kubectl/blob/197123726db24c61aa0f78d1f0ba6e91a2ec2f35/pkg/cmd/apply/apply.go#L564-L602
+
+## Backwards compatibility
+
+<!-- Describe potential impact and severity on pre-existing code. -->
+
+SSA should be generally compatible functionality-wide with CSA.
+The expectation is that SSA should generally produce equivalent Kubernetes objects as CSA.
+
+There four main behavioural differences for Helm switching to SSA:
+
+1. The differences between object's manifests managed with SSA vs the existing three-way strategic merge process:
+   - Arrays will be merged
+   - Removal of unset fields
+   - Default values
+2. Issues with a user enabling/disabling SSA over the lifetime of a release e.g. environment differences or older Helm clients which don't understand SSA
+3. SSA will apply to custom resources
+4. Field management ownership conflicts
+
+For 1., chart-operators may opt-out out of SSA.
+For 2., the user can control by ensuring SSA is consistently used within their environment.
+For 3., the it is generally considered a limitation of Helm not updating custom resources, for which SSA is expected to be an improvement.
+For 4., the `--force-conflicts` flag exists.
+
+Additionally as noted, SSA isn't compatible with very old Kubernetes versions (Kubernetes v1.17 and prior)
```

Comment:
You said 1.22, earlier. 🤷

Reply:
GA in 1.22

Iirc, SSA will work via beta release from 1.17. At this point, 1.17 vs 1.22 is probably water under the bridge now. Will improve this slightly.